simtools/ray_tracing.py

```diff
@@ -59,110 +66,106 @@ def __init__(
-            label=self.label,
+
+    def __repr__(self):
+        """Return string representation of RayTracing class."""
+        return f"RayTracing(label={self.label})\n"
+
+    def _initialize_config(
+        self,
+        zenith_angle,
+        off_axis_angle,
+        source_distance,
+        single_mirror_mode,
+        use_random_focal_length,
+        mirror_numbers,
+    ):
+        """Initialize ray tracing configuration to namedtuple."""
+        config_data = namedtuple(
+            "Config",
+            [
+                "zenith_angle",
+                "off_axis_angle",
+                "source_distance",
+                "single_mirror_mode",
+                "use_random_focal_length",
+                "mirror_numbers",
+            ],
+        )
+        if single_mirror_mode:
+            source_distance, mirror_numbers = self._initialize_single_mirror_mode(mirror_numbers)
+        # round the off-axis angles so the values in results table are the same as provided.
+        off_axis_angle = np.around(off_axis_angle, 5)
+
+        return config_data(
+            zenith_angle=zenith_angle.to("deg").value,
+            off_axis_angle=off_axis_angle.to("deg").value,
+            source_distance=source_distance.to("km").value,
+            single_mirror_mode=single_mirror_mode,
+            use_random_focal_length=use_random_focal_length,
+            mirror_numbers=mirror_numbers,
         )
-        self._output_directory.joinpath("results").mkdir(parents=True, exist_ok=True)
-        self._file_results = self._output_directory.joinpath("results").joinpath(file_name_results)
 
-    @classmethod
-    def from_kwargs(cls, **kwargs):
+    def _initialize_single_mirror_mode(self, mirror_numbers):
         """
-        Build a RayTracing object from kwargs only.
-
-        The configurable parameters can be given as kwargs, instead of using the
-        config_data or config_file arguments.
+        Initialize single mirror mode.
 
         Parameters
         ----------
-        kwargs
-            Containing the arguments and the configurable parameters.
+        mirror_numbers: list, str
+            List of mirror numbers (or 'all').
 
         Returns
         -------
-        Instance of this class.
+        tuple
+            Tuple containing source distance and mirror numbers.
         """
-        args, config_data = gen.separate_args_and_config_data(
-            expected_args=[
-                "telescope_model",
-                "label",
-                "simtel_path",
-            ],
-            **kwargs,
+        self._logger.debug(
+            "Single mirror mode is activated - "
+            "source distance is being recalculated to 2 * focal length"
         )
-        return cls(**args, config_data=config_data)
+        mir_focal_length = self.telescope_model.get_parameter_value("mirror_focal_length")
+        source_distance = 2 * float(mir_focal_length) * u.cm.to(u.km) * u.km
```

Comment:
It seems like there are too many `u.km` in this line.

Reply:
Help me with this:

- `u.cm.to(u.km)` is converting the focal length (in cm) to km.
- `*u.km` makes sure that source distance is an astropy quantity with km as unit.

I don't see that I can remove one of them.

Reply:
I expected that `2 * float(mir_focal_length) * u.cm.to(u.km)` would already give you an astropy quantity. I expected that would be the difference between `to_value(u.km)`. But I guess that's not the case here. 
Either way, I think it would be clearer to write instead:

```
(2 * float(mir_focal_length) * u.cm).to(u.km)
```
to get a quantity.

Reply:
This is indeed clearer. I've changed it accordingly.